File: app/channels/community_channel.rb
class CommunityChannel < ApplicationCable::Channel
  def subscribed
    stream_from "community_channel_#{params['community']}"
  end

  def unsubscribed
    # Any cleanup needed when channel is unsubscribed
  end

  def speak(data)
    Message.create! content: data['message'], user_id: current_user.id, community_id: params['community']
  end
end

File: app/javascript/channels/community_channel.js
import consumer from "./consumer"

$(function() {
  const chatChannel = consumer.subscriptions.create({ channel: 'CommunityChannel', room: $('#messages').data('community_id') }, {
    connected() {
      // Called when the subscription is ready for use on the server
    },

    disconnected() {
      // Called when the subscription has been terminated by the server
    },

    received: function(data) {
      return $('#messages').append(data['message']);
    },

    speak: function(message) {
      return this.perform('speak', {
        message: message
      });
    }
  });

  $(document).on('keypress', '[data-behavior~=community_speaker]', function(event) {
    if (event.key === 'Enter') {
      chatChannel.speak(event.target.value);
      event.target.value = '';
      return event.preventDefault();
    }
  });
});